molgenis-data/src/main/java/org/molgenis/data/meta/model/AttributeMetaData.java

```diff
@@ -60,46 +60,21 @@ public AttributeMetaData(String attrId, EntityMetaData entityMeta)
 	}
 
 	/**
-	 * Copy-factory (instead of copy-constructor to avoid accidental method overloading to {@link #AttributeMetaData(EntityMetaData)})
+	 * Copy-factory (instead of copy-constructor to avoid accidental method overloading to
+	 * {@link #AttributeMetaData(EntityMetaData)}). Creates a copy of attribute with a shallow copy of referenced
+	 * entity and tags.
 	 *
-	 * @param attrMeta attribute
+	 * @param attrMeta     attribute
+	 * @param attrCopyMode attribute copy mode that defines whether to deep-copy or shallow-copy attribute parts
 	 * @return deep copy of attribute
 	 */
-	public static AttributeMetaData newInstance(AttributeMetaData attrMeta)
+	public static AttributeMetaData newInstance(AttributeMetaData attrMeta, AttributeCopyMode attrCopyMode)
 	{
-		return newInstance(attrMeta, Maps.newConcurrentMap());
-	}
-
-	/**
-	 * Copy-factory (instead of copy-constructor to avoid accidental method overloading to {@link #AttributeMetaData(EntityMetaData)})
-	 *
-	 * @param attrMeta attribute
-	 * @return deep copy of attribute
-	 */
-	static AttributeMetaData newInstance(AttributeMetaData attrMeta, Map<String, StaticEntity> copied)
-	{
-		if(null == attrMeta) {
-			return null;
-		}
-
-		requireNonNull(attrMeta.getName());
-
-		if(copied.containsKey(attrMeta.getName()))
-		{
-			return (AttributeMetaData) copied.get(attrMeta.getName());
-		}
-
-		EntityMetaData entityMeta = attrMeta.getEntityMetaData();
-		AttributeMetaData attrMetaCopy = new AttributeMetaData(entityMeta);
+		AttributeMetaData attrMetaCopy = new AttributeMetaData(attrMeta.getEntityMetaData()); // do not deep-copy
```

Comment:
The AttributeMetaData should not refer to existing attributes right? So why not deep copy?